File: src/com/quran/labs/androidquran/util/QuranScreenInfo.java
package com.quran.labs.androidquran.util;

import android.content.res.Configuration;

public class QuranScreenInfo {

	private static QuranScreenInfo instance = null;
	
	private int width;
	private int height;
	private int max_width;
	private int orientation;
	
	private QuranScreenInfo(int width, int height){
		this.orientation = Configuration.ORIENTATION_PORTRAIT;
		this.width = width;
		this.height = height;
		this.max_width = (width > height)? width : height;
	}
	
	public static QuranScreenInfo getInstance(){
		return instance;
	}
	
	public static void initialize(int width, int height){
		instance = new QuranScreenInfo(width, height);
	}
	
	public int getWidth(){ return this.width; }
	public int getHeight(){ return this.height; }
	
	public String getWidthParam(){
		if (this.max_width <= 480) return "_480";
		else if (this.max_width <= 800) return "_800";
		else return "_1024";
	}
	
	public boolean isLandscapeOrientation() {
		return this.orientation == Configuration.ORIENTATION_LANDSCAPE;
	}
		
	public void setOrientation(int orientation) {
		this.orientation = orientation;
	}
}

File: app/src/main/java/com/quran/labs/androidquran/util/QuranScreenInfo.kt
package com.quran.labs.androidquran.util

import android.content.Context
import android.graphics.Point
import android.view.Display
import com.quran.data.source.PageSizeCalculator
import com.quran.mobile.di.qualifier.ApplicationContext
import timber.log.Timber
import javax.inject.Inject
import kotlin.math.max

class QuranScreenInfo @Inject constructor(
  @ApplicationContext appContext: Context,
  display: Display,
  pageSizeCalculator: PageSizeCalculator
) {
  private val height: Int
  private val altDimension: Int
  private val maxWidth: Int
  private val orientation: Int
  private val context: Context
  private val pageSizeCalculator: PageSizeCalculator

  init {
    val point = Point()
    display.getSize(point)

    height = point.y
    altDimension = point.x
    maxWidth = max(point.x.toDouble(), point.y.toDouble()).toInt()
    orientation = appContext.resources.configuration.orientation

    this.context = appContext
    this.pageSizeCalculator = pageSizeCalculator
    Timber.d("initializing with %d and %d", point.y, point.x)
  }

  fun getHeight(): Int {
    return if (orientation == context.resources.configuration.orientation) {
      height
    } else {
      altDimension
    }
  }

  val widthParam: String
    get() {
      pageSizeCalculator.setOverrideParameter(
        QuranSettings.getInstance(context).defaultImagesDirectory
      )
      return "_" + pageSizeCalculator.getWidthParameter()
    }

  val tabletWidthParam: String
    get() {
      pageSizeCalculator.setOverrideParameter(
        QuranSettings.getInstance(context).defaultImagesDirectory
      )
      return "_" + pageSizeCalculator.getTabletWidthParameter()
    }

  val isDualPageMode: Boolean
    get() = maxWidth > 960
}
